activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -11,12 +11,10 @@ def collection_cache_key(collection = all, timestamp_column = :updated_at) # :no
         end
       else
         column_type = type_for_attribute(timestamp_column.to_s)
-        column = "#{connection.quote_table_name(collection.table_name)}.#{connection.quote_column_name(timestamp_column)}"
+        column = "#{connection.quote_table_name("timestamps")}.#{connection.quote_column_name(timestamp_column)}"
 
-        query = collection
-          .unscope(:select)
-          .select("COUNT(*) AS #{connection.quote_column_name("size")}", "MAX(#{column}) AS timestamp")
-          .unscope(:order)
+        query = "SELECT COUNT(*) AS #{connection.quote_column_name("size")}, MAX(#{column}) AS timestamp" <<
```

Comment:
I believe this implementation will blow up if called on a query that contains joins. Can you add a test to verify?